src/app/CommandHandler.h

```diff
@@ -50,8 +53,33 @@ class CommandHandler : public Command
 
 private:
     friend class TestCommandInteraction;
+    friend class CommandHandlerHandleDeleter;
     CHIP_ERROR SendCommandResponse();
     CHIP_ERROR ProcessCommandDataElement(CommandDataElement::Parser & aCommandElement) override;
+    static void FinishPendingWork(System::Layer * aLayer, void * aAppState, CHIP_ERROR aError);
+
+    struct CommandHandlerHandleDeleter
+    {
+        void operator()(CommandHandler * apCommandHandler) const;
+    };
+
+    uint32_t mPendingWorkCount = 0;
+
+public:
+    using CommandHandlerAsyncHandle = std::unique_ptr<CommandHandler, CommandHandlerHandleDeleter>;
```

Comment:
Is `std::unique_ptr` safe for use on embedded platforms? Last I checked, it didn't look like it was used anywhere in embedded code?